pyccel/codegen/pipeline.py

```diff
@@ -212,14 +218,16 @@ def handle_error(stage):
         handle_error('parsing (syntax)')
         raise PyccelSyntaxError('Syntax step failed')
 
+    end_syntax = time.time()
+
     if syntax_only:
         pyccel_stage.pyccel_finished()
         return
```

Comment:
Don't we have a problem here? We are not printing any timings unless we get to the end of this function.

As a workaround I suggest that we update the `timings` dictionary on the fly (the overhead of doing so should be negligible compared to everything else that goes on in Pyccel) and then pass it to a `print_timings` function which prints whatever is available and the total elapsed time.

Reply:
No but I think documenting it will be :laughing:

Reply:
Can I put it as a function in a function?

Reply:
Sure